internal/xdscache/v3/listener.go

```diff
@@ -156,6 +159,122 @@ type RateLimitConfig struct {
 	EnableXRateLimitHeaders bool
 }
 
+func NewListenerConfig(
```

Comment:
I commented this at the calling site, but seeing the function definition really drives home that this one really should take the config object and extract the fields itself. Not blocking for this one, but should be refactored at some point.

Reply:
yeah I agree the params aren't great, but I don't think passing the full ContourConfig.Spec is good either. A future place to refactor for sure. =)